inc/html.class.php

```diff
@@ -5770,6 +5789,96 @@ static function textarea($options = []) {
+    *    - uploads             array    uploads to display (done in a previous form submit)
+    * @return void|string   the html if display parameter is false
+    */
+   private static function uploadedFiles($options = []) {
+      global $CFG_GLPI;
+
+      //default options
+      $p['filecontainer']     = 'fileupload_info';
+      $p['name']              = 'filename';
+      $p['editor_id']         = '';
+      $p['display']           = true;
+      $p['uploads']           = [];
+
+      //merge default options with options parameter
+      $p = array_merge($p, $options);
+
+      // div who will receive and display file list
+      $display = "<div id='".$p['filecontainer']."' class='fileupload_info'>";
+      if (isset($p['uploads']['_' . $p['name']])) {
+         foreach ($p['uploads']['_' . $p['name']] as $uploadId => $upload) {
+            $prefix  = substr($upload, 0, 23);
+            $displayName = substr($upload, 23);
+
+            // get the extension icon
+            $extension = pathinfo(GLPI_TMP_DIR . '/' . $upload, PATHINFO_EXTENSION);
+            $extensionIcon = '/pics/icones/' . $extension . '-dist.png';
+            if (!is_readable(GLPI_ROOT . $extensionIcon)) {
+               $extensionIcon = '/pics/icones/defaut-dist.png';
+            }
+            $extensionIcon = $CFG_GLPI['root_doc'] . $extensionIcon;
+
+            $upload = [
+               'name'    => $upload,
+               'id'      => 'doc' . $p['name'] . mt_rand(),
+               'display' => $displayName,
+               'size'    => filesize(GLPI_TMP_DIR . '/' . $upload),
+               'prefix'  => $prefix,
+            ];
+            $tag = $p['uploads']['_tag_' . $p['name']][$uploadId];
+            $tag = [
+               'name' => $tag,
+               'tag'  => "#$tag#",
+            ];
+
+            // Show the name and size of the upload
+            $display .= "<p id='" . $upload['id'] . "'>&nbsp;";
+            $display .= "<img src='$extensionIcon' title='$extension'>&nbsp;";
+            $display .= "<b>" . $upload['display'] . "</b>&nbsp;(" . Toolbox::getSize($upload['size']) . ")";
+
+            $name = '_' . $p['name'] . '[' . $uploadId . ']';
+            $display .= Html::hidden($name, ['value' => $upload['name']]);
+
+            $name = '_prefix_' . $p['name'] . '[' . $uploadId . ']';
+            $display .= Html::hidden($name, ['value' => $upload['prefix']]);
+
+            $name = '_tag_' . $p['name'] . '[' . $uploadId . ']';
+            $display .= Html::hidden($name, ['value' => $tag['name']]);
+
+            // show button to delete the upload
+            $getEditor = 'null';
+            if ($p['editor_id'] != '') {
+               $getEditor = "tinymce.get('" . $p['editor_id'] . "')";
+            }
+            $textTag = $tag['tag'];
+            $domItems = "{0:'" . $upload['id'] . "', 1:'" . $upload['id'] . "'+'2'}";
+            $deleteUpload = "deleteImagePasted($domItems, '$textTag', $getEditor)";
```

Comment:
It works by just passing only one element. What is this `'" . $upload['id'] . "'+'2'` ?
```suggestion
            $deleteUpload = "deleteImagePasted(['{$upload['id']}'], '$textTag', $getEditor)";
```

Reply:
This method mimics the HTML snippet produced by displayUploadedFile() in fileupload.js . I did not found the reason of this concatenation in the original JS code.